sdk/storage/azure-storage-blob/azure/storage/blob/_shared/streams.py

```diff
@@ -160,7 +253,7 @@ def read(self, size: int = -1) -> bytes:
                     SMRegion.MESSAGE_FOOTER):
                 count += self._read_metadata_region(self._current_region, remaining, output)
             elif self._current_region == SMRegion.SEGMENT_CONTENT:
-                count += self._read_content(size, output)
+                count += self._read_content(remaining, output)
```

Comment:
This was a bug that somehow snuck through testing before. I augmented the tests to catch this.